zapcore/write_syncer.go

```diff
@@ -75,6 +77,62 @@ func (s *lockedWriteSyncer) Sync() error {
 	return err
 }
 
+type bufferWriterSyncer struct {
+	bufferWriter *bufio.Writer
+	ws           WriteSyncer
+}
+
+// defaultBufferSize sizes the buffer associated with each log file. It's large
+// so that log records can accumulate without the logging thread blocking
+// on disk I/O untill buffer fills up. The flushDaemon will block instead.
+const defaultBufferSize = 256 * 1024
+
+// defaultFlushInterval means the default flush interval
+const defaultFlushInterval = 30 * time.Second
+
+// Buffer wraps a WriteSyncer in a buffer to improve performance,
```

Comment:
Add comments about periodic flushing here, and mention that if 0 is passed in, what the defaults used are (can just reference the constant names)

Reply:
done

Reply:
Looking at the changes to `Write`, it looks like we're trying to guarantee that a single `Write` is not split up -- but we may want to mention here that multiple `Write` calls will be batched. Just in case there's some write syncers which assume that a `Write` only contains a single log entry, which isn't compatible with buffering.

Reply:
I think the above is still pending.

We should also call out the behaviour on errors in `Sync`

Reply:
> we may want to mention here that multiple `Write` calls will be batched

I don't think we documented this anywhere, and it's pretty important

Reply:
Sure, I'll add some comments here.